blocks/extensions/plot.py

```diff
@@ -136,14 +132,86 @@ def preexec_fn():
             logger.info('Plotting server PID: {}'.format(self.sub.pid))
         else:
             self.sub = None
-        output_server(self.document, url=self.server_url)
 
-    def __getstate__(self):
-        state = self.__dict__.copy()
-        state['sub'] = None
-        return state
+        self.session = Session(
+            root_url=self.server_url,
+            load_from_config=False)
+        output_server(self.document, session=self.session)
+
+
+class PlotHistogram(SimpleExtension):
+
+    def __init__(self, document, channels, server_url='default', **kwargs):
+        """ Live plot an histogram of mean and std dev of the provided channels
+
+        This extension live plots an histogram of the mean value of each of the
+        provided channels over the monitored dataset. It will also display
+        confidence intervals representing the standard deviation of the channel
+        over the dataset.
+
+        .. warning::
+            This extension assumes that a bokeh server is running at the
+            provided url and won't try to start a server autonomously. To
+            start a bokeh server run `bokeh-server --ip=<ip> --port=<port>`.
+            If you want the server to accept connections from any address use
+            <ip>= 0.0.0.0.
+
+        Parameter
+        ---------
+        document : string
+            The name of the document.
+        channels : list
+            The channels to be displayed as a histogram.
+        server_url : string
+            The url of the server
+        """
+        self.document = document
+        self.channels = channels
+        self.server_url = server_url
 
-    def __setstate__(self, state):
```

Comment:
I see. In this case I will restore them and also implement the equivalent for `HistogramPlot` too.